ext/date/php_date.c

```diff
@@ -4224,7 +4212,90 @@ PHP_METHOD(DatePeriod, __construct)
 
 	dpobj->initialized = 1;
 }
-/* }}} */
+
+PHP_METHOD(DatePeriod, createFromRecurrences)
+{
+	zval *start, *interval;
+	zend_long recurrences, options = 0;
+	zval object;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "OOl|l", &start, date_ce_interface, &interval, date_ce_interval, &recurrences, &options) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	object_init_ex(&object, zend_get_called_scope(execute_data));
+
+	construct_date_period(INTERNAL_FUNCTION_PARAM_PASSTHRU, &object, start, NULL, interval, recurrences, options, NULL, 0);
+	if (EG(exception)) {
+		zval_ptr_dtor(&object);
+		RETURN_THROWS();
+	}
+
+	ZVAL_COPY_VALUE(return_value, &object);
+}
+
+PHP_METHOD(DatePeriod, createFromDates)
```

Comment:
other candidates I thought about: `createBetweenDates`, `createBetweenStartAndEnd`, `createFromStartAndEnd`

Reply:
I think I would prefer `createFromStartAndEnd`, and I would also switch the order of arguments to: 
```
    public static function createFromStartAndEnd(DateTimeInterface $start, DateTimeInterface $end, DateInterval $interval, int $options = 0): static {}
```

Reply:
Another suggestion would be to use `createFromEndpoints` or `createFromDatepoints` to imply that the `DatePeriod` endpoints are expressed using `DateTimeInterface` objects.